kivy/app.py

```diff
@@ -654,7 +662,25 @@ def get_application_config(self):
         '''
 
         if platform == 'android':
-            defaultpath = '/sdcard/.%(appname)s.ini'
+            from jnius import autoclass
+            activity = autoclass('org.kivy.android.PythonActivity').mActivity
+            defaultpath = join(
+                activity.getFilesDir().getAbsolutePath(),
+                'app',
```

Comment:
Perhaps we should omit the 'app', that way it isn't in the main application files directory (which is fine), and also will work if anyone happens to use it with old p4a versions.

Reply:
Updated comment: I'm pretty sure we should omit the `'app'`. This should mean that the config does not get overwritten on app update. This is actually quite important.

Reply:
@inclement I'm on it.